src/app/services/resource.service.ts

```diff
@@ -259,4 +261,36 @@ export class ResourceService {
       });
   }
 
+  /**
+   * Gets all elements from the study dimension that satisfy the constaint if given
+   * @param {Constraint} constraint
+   * @returns {Observable<string[]>}
+   */
+  getStudyNames(constraint: Constraint): Observable<string[]> {
+    return this.transmartResourceService.getStudyNames(constraint)
+      .map((elements: TransmartStudyDimensionElements[]) => {
```

Comment:
'TransmartStudyDimensionElements' is plural? is that correct?

Reply:
ah... no, I'll change that